mlrun/runtimes/remotesparkjob.py

```diff
@@ -129,6 +131,11 @@ def spec(self, spec):
 
     def with_spark_service(self, spark_service, provider=RemoteSparkProviders.iguazio):
         """Attach spark service to function"""
+        # check for existence of spark_services
+        if not get_k8s_helper().is_service_exist(name=spark_service):
+            raise MLRunInvalidArgumentError(
+                f"spark service named {spark_service} doesn't exist"
```

Comment:
Be nicer in the error message :)

`The spark service provided '{}' does not exist. Please notice the name should be given as it is shown in the Iguazio UI.`

Something like that